Letterbook.Adapter.TimescaleFeeds/EntityModels/TimelinePost.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.Diagnostics;
+using Letterbook.Core.Models;
+
+namespace Letterbook.Adapter.TimescaleFeeds.EntityModels;
+
+/// <summary>
+/// A partial view of a Post, adapted for use and query performance in Timelines
+/// </summary>
+[DebuggerDisplay("PostId: {PostId}, AudienceId: {AudienceId}")]
+public class TimelinePost
+{
+	/// <summary>
+	/// Time is the timeseries key. TSDBs will time-order everything and do rapid time-indexed queries.
+	/// In our case, records should generally have a time value of right now, because what we care about is when it got added to
+	/// the timeline. When it was added could and likely will be different from any of the datetime properties on Post
+	/// </summary>
+	public DateTimeOffset Time { get; set; } = DateTimeOffset.UtcNow;
+
+	/// <summary>
+	/// Id of the Post
+	/// </summary>
+	public required Guid PostId { get; set; }
+
+	/// <summary>Same as <see cref="Post"/></summary>
+	public required string Preview { get; set; }
+
+	/// <summary>Same as <see cref="Post"/></summary>
+	public required string Authority { get; set; }
+
+	/// <summary>Same as <see cref="Post"/></summary>
+	public List<TimelineProfile> Creators { get; set; } = new();
+
+	/// <summary>Same as <see cref="Post"/></summary>
+	public string? Summary { get; set; }
+
+	/// <summary>Same as <see cref="Post"/></summary>
+	public required DateTimeOffset UpdatedDate { get; set; }
+
+	/// PostId of the InReplyTo Post <summary>Same as <see cref="Post"/></summary>
+	public Guid? InReplyToId { get; set; }
+
+	/// <summary>
+	/// Posts can be shared by multiple profiles, but each share would result in at-most one addition to the timeline.
+	/// So TimelinePost has a single SharedBy property
+	/// </summary>
+	public TimelineProfile? SharedBy { get; set; }
+
+	/// <summary>
+	/// Posts can, and often will, be available to multiple Audiences. Posts with multiple Audiences will be duplicated
+	/// in the Timeline table, with one record for each Audience.
+	/// This allows us to query for specific memberships using WHERE IN (list) clauses, rather than requiring joins or
+	/// complex set intersections
+	/// </summary>
+	public required Uri AudienceId { get; set; }
+
+	/// <summary>ID of the Thread to which the Post belongs</summary>
+	public required Guid ThreadId { get; set; }
+
+	public static implicit operator Post(TimelinePost p) => new Post
+	{
+		Id = p.PostId,
+		Thread = new ThreadContext()
+		{
+			Id = p.ThreadId,
+			FediId = default!,
```

Comment:
is it safe to forcibly set this non-nullable field to null?

Reply:
Yeah. Safe-ish, anyway. The only instance where this constructor should be used is when we already have a thread ID, and we just need to look it up to fill out the rest of the object.

It would be a bug for an object created this way to be written to the db, and the query would fail. So, it would produce some error messages and some action would fail, but I don't think it could corrupt any data.